### PR TITLE
Cleaning up generated HDL code

### PR DESCRIPTION
**Pull request by BFH-ktt1**

### CODE REVIEW

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java [hunk 1]
```diff
@@ -74,82 +73,43 @@ public boolean generateAllHDLDescriptions(
...
-        oneLine.append(": ").append(typedWires.get(wire)).append(";");
-        contents.add("   SIGNAL {{1}}", oneLine.toString());
-        oneLine.setLength(0);
+        maxNameLength = Math.max(maxNameLength, wire.length());
+        mySignals.put(wire, typedWires.get(wire));
       }
-      contents.add("")
-          .add("BEGIN")
+      // now we add them
+      if (maxNameLength > 0) contents.addRemarkBlock("Here all used signals are defined");
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java:106]: What about `All used signals are defined here` instead?

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

**Review by MarcinOrlowski — APPROVED** [review]

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java (hunk 1 above)
Comment (maehne) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java:104]: Personally, I don't see much value in this comment. It just restates, what the code already expresses.

**Review by maehne — DISMISSED** [review]
Looks so far good to me.

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java (hunk 1 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java:104]: Not quite, as also the I/O components and in-lined components looks like "wiring". I prefer to be verbose here for both debugging purposes, as well as being "transparent" in the generated code

**Review by maehne — APPROVED** [review]

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java (hunk 1 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java]: fixed

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7432.java [hunk 2]
```diff
@@ -26,12 +25,9 @@
...
 
     @Override
-    public ArrayList<String> getLogicFunction(int index) {
-      final var contents = new ArrayList<String>();
-      contents.add("   " + Hdl.assignPreamble() + "gate_" + index + "_O" + Hdl.assignOperator()
-              + "gate_" + index + "_A" + Hdl.orOperator() + "gate_" + index + "B;");
-      contents.add("");
-      return contents;
+    public LineBuffer getLogicFunction(int index) {
+      return LineBuffer.getHdlBuffer()
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl7432.java:29]: BTW: would it be easier if HDL placeholders were always available (so just `getBuffer()` and you all set)?

File: src/main/java/com/cburch/logisim/std/gates/AbstractGateHDLGenerator.java [hunk 3]
```diff
@@ -81,24 +80,24 @@ public boolean getFloatingValue(boolean isInverted) {
...
     contents.add(getLogicFunction(nrOfInputs, bitWidth, onehot));
-    return contents.get();
+    return contents;
   }
 
-  public ArrayList<String> GetOneHot(boolean inverted, int nrOfInputs, boolean isBus) {
-    var lines = new ArrayList<String>();
-    var spaces = "   ";
+  public LineBuffer getOneHot(boolean inverted, int nrOfInputs, boolean isBus) {
+    var lines = LineBuffer.getHdlBuffer();
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/gates/AbstractGateHDLGenerator.java:87]: `final`

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7432.java (hunk 2 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/ttl/Ttl7432.java:29]: No, I actually like the differentiation between ```getBuffer```, and  ```getVhdlBuffer``` as it (1) makes LineBuffer more versatile, and (2) it is more readable as in the second case we know we are going to write HDL code :-)

File: src/main/java/com/cburch/logisim/std/memory/TFlipFlop.java [hunk 4]
```diff
@@ -36,10 +35,10 @@ public TFFHDLGeneratorFactory() {
     }
 
     @Override
-    public ArrayList<String> GetUpdateLogic() {
-      return LineBuffer.getBuffer()
-          .add("{{1}} s_next_state {{2}} s_current_state_reg {{3}} T;", Hdl.assignPreamble(), Hdl.assignOperator(), Hdl.xorOperator())
-          .getWithIndent();
+    public LineBuffer getUpdateLogic() {
+      return LineBuffer.getHdlBuffer()
+          .pair("xor", Hdl.xorOperator())
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/memory/TFlipFlop.java:40]: Maybe add that pair to LineBuffer's `addHdlPairs()` directly?

File: src/main/java/com/cburch/logisim/std/ttl/Ttl74266.java [hunk 5]
```diff
@@ -26,12 +26,10 @@
...
     @Override
-    public ArrayList<String> getLogicFunction(int index) {
-      final var contents = new ArrayList<String>();
-      contents.add("   " + Hdl.assignPreamble() + "gate_" + index + "_O" + Hdl.assignOperator() + Hdl.notOperator()
-              + "(gate_" + index + "_A" + Hdl.xorOperator() + "gate_" + index + "_B);");
-      contents.add("");
-      return contents;
+    public LineBuffer getLogicFunction(int index) {
+      return LineBuffer.getHdlBuffer()
+          .pair("xor", Hdl.xorOperator())
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl74266.java:31]: If `xor` is added to `addHdlPairs()` this one can be removed.

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7430HDLGenerator.java [hunk 6]
```diff
@@ -34,12 +32,9 @@ public Ttl7430HDLGenerator() {
...
   @Override
-  public ArrayList<String> getModuleFunctionality(Netlist TheNetlist, AttributeSet attrs) {
-    final var contents = new ArrayList<String>();
-    contents.add("   " + Hdl.assignPreamble() + "Y1" + Hdl.assignOperator() + Hdl.notOperator() + "(A"
-            + Hdl.andOperator() + "B" + Hdl.andOperator() + "C" + Hdl.andOperator() + "D" + Hdl.andOperator() + "E"
-            + Hdl.andOperator() + "F" + Hdl.andOperator() + "G" + Hdl.andOperator() + "H);");
-    return contents;
+  public LineBuffer getModuleFunctionality(Netlist TheNetlist, AttributeSet attrs) {
+    return LineBuffer.getHdlBuffer()
+        .add("{{assign}}Y{{=}}{{not}}(A{{and}}B{{and}}C{{and}}D{{and}}E{{and}F{{and}}}G{{and}}H):");
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl7430HDLGenerator.java:37]: `}}}`

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7486.java [hunk 7]
```diff
@@ -26,12 +26,10 @@
...
     @Override
-    public ArrayList<String> getLogicFunction(int index) {
-      final var contents = new ArrayList<String>();
-      contents.add("   " + Hdl.assignPreamble() + "gate_" + index + "_O" + Hdl.assignOperator()
-              + "gate_" + index + "_A" + Hdl.xorOperator() + "gate_" + index + "_B;");
-      contents.add("");
-      return contents;
+    public LineBuffer getLogicFunction(int index) {
+      return LineBuffer.getHdlBuffer()
+          .pair("xor", Hdl.xorOperator())
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl7486.java:31]: If xor is added to addHdlPairs() this one can be removed.

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/std/memory/TFlipFlop.java (hunk 4 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/memory/TFlipFlop.java:40]: I agree, this one is missing

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7430HDLGenerator.java (hunk 6 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/ttl/Ttl7430HDLGenerator.java:37]: Thanks! Fixed

File: src/main/java/com/cburch/logisim/std/gates/AbstractGateHDLGenerator.java (hunk 3 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/gates/AbstractGateHDLGenerator.java:87]: Done

File: src/main/java/com/cburch/logisim/std/memory/TFlipFlop.java (hunk 4 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/memory/TFlipFlop.java:40]: Added

File: src/main/java/com/cburch/logisim/std/ttl/Ttl74266.java (hunk 5 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/ttl/Ttl74266.java:31]: Done

File: src/main/java/com/cburch/logisim/std/ttl/Ttl7486.java (hunk 7 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/ttl/Ttl7486.java:31]: Done

File: src/main/java/com/cburch/logisim/std/arith/ShifterHDLGeneratorFactory.java [hunk 8]
```diff
@@ -124,10 +124,10 @@ public void getGenerationTimeWiresPorts(Netlist theNetlist, AttributeSet attrs)
             """, nrOfShiftBits - 1);
       }
     }
-    return contents.getWithIndent();
+    return contents;
   }
 
-  private ArrayList<String> GetStageFunctionalityVerilog(int stageNumber, int nrOfBits) {
+  private ArrayList<String> getStageFunctionalityVerilog(int stageNumber, int nrOfBits) {
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/arith/ShifterHDLGeneratorFactory.java:130]: Just wonder if it'd not be more consistent to have this (`getStageFunctionalityVhdl()` too) return LineBuffer too?

File: src/main/java/com/cburch/logisim/std/io/DotMatrixHdlGeneratorFactory.java [hunk 9]
```diff
@@ -100,7 +88,7 @@
     }
     contents.add("");
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/io/DotMatrixHdlGeneratorFactory.java:89]: `contents.empty()`?

File: src/main/java/com/cburch/logisim/std/wiring/AbstractConstantHdlGeneratorFactory.java [hunk 10]
```diff
@@ -24,39 +23,35 @@ public long getConstant(AttributeSet attrs) {
...
-        Contents.add("{{assign}} {{1}} {{=}} {{2}};", Hdl.getNetName(componentInfo, 0, true, nets), Hdl.getConstantVector(ConstantValue, 1))
+        contents.add("{{assign}} {{1}} {{=}} {{2}};", Hdl.getNetName(componentInfo, 0, true, nets), Hdl.getConstantVector(constantValue, 1))
             .add("");
       } else {
         if (nets.isContinuesBus(componentInfo, 0)) {
           /* easy case */
-          Contents.add("{{assign}} {{1}} {{=}} {{2}};", Hdl.getBusNameContinues(componentInfo, 0, nets), Hdl.getConstantVector(ConstantValue, NrOfBits));
-          Contents.add("");
+          contents.add("{{assign}} {{1}} {{=}} {{2}};", Hdl.getBusNameContinues(componentInfo, 0, nets), Hdl.getConstantVector(constantValue, nrOfBits));
+          contents.add("");
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/wiring/AbstractConstantHdlGeneratorFactory.java:39]: `empty()` + I'd chain this one.

**Review by MarcinOrlowski — APPROVED** [review]

File: src/main/java/com/cburch/logisim/std/arith/ShifterHDLGeneratorFactory.java (hunk 8 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/arith/ShifterHDLGeneratorFactory.java:130]: I'll look into it in a new PR, I do not want to block yours more

Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/std/arith/ShifterHDLGeneratorFactory.java:130]: It was just a note/comment, hence I approved regardless of your answer.

**Pull request merged**